### ISSUE TITLE
Question: MSAL Auth: cookie needed when using WKWebView

### ISSUE THREAD

**Issue by renep** [AzureAD/microsoft-authentication-library-for-objc#1906]
I have integrated the MSAL framework to the App I'm developing. The goal is to have a single sign and connect to a internal server. For this the Azure App Proxy is used.
Then authentication works and also the connection to the server work. But for every request the "Authorization" header must be set with the "Bearer" token.
The Problem now is that the App also uses WKWebViews where the content is loaded from the server. For the initial request the "Authentication" header can be set, but the resulting HTML page contains <script> and <style> tags, that must be loaded, but the WKWebView load these without the "Authentication" header. Therefor the website cannot be loaded properly.
So for this to work a cookie is needed. Is it possible to get a proper AzureAppProxyAccessCookie cookie for this?

**Comment by swasti29** [comment]
@renep It is possible for the resulting page to reattch the "Bearer" token ? Intead of " WKWebViews" you can use "aswebauthenticationsession". For more details see [here](https://learn.microsoft.com/en-us/entra/identity-platform/customize-webviews#in-app-browser)

**Comment by renep** [comment]
@swasti29 This article only describes how to change the default browser for the authenication requests. The authentication itself is not the problem here.

**Comment by swasti29** [comment]
@renep We do not support cookie based authentication.
In general, token based authentication is recommended over cookie based authentication.
Also, Can you clarify the problem in details? 
Is the objective to have session cookie instead of Bearer Token upon successful authentication flow with identity provider and client app uses the same in subsequent requests?

**Comment by renep** [comment]
The authentication itself works fine. Also with the token. This is not the problem. The App I'm integrating the MSAL framework uses most of the time JSON-Rest request to communicate with the server. This requests are not the problem, because here I can add the Bearer Token when I want to access a server via the Azure App Proxy. But the app also uses WKWebViews and HTML sites were are loaded form the server. We use the HTML pages for forms and similar stuff. But this website does not load properly, because for the inital request I can add the Bearer Token, but the html page contains also addination resources like scripts and css resource, and the WKWebView tries to load this resources from the server, but can't, because the WKWebView does not know anything about the Bearer Token. Here the WKWebView needs a cookie. 
E.g. Our Server also provide a Web Interface that can be access through the Browser. Here the Azure AD Authentication with App Proxy also works, but a AzureAppProxyAccessCookie is set, so that the communication with the server via the App Proxy works.
So at my point of view, I need the ability to set a cookie for the WKWebView, so that the requests can go through the App Proxy.

**Comment by swasti29** [comment]
@renep 
I am not aware of the AzureAppProxyAccessCookie cookie. 
As per the aticle given [here](https://learn.microsoft.com/en-us/entra/identity-platform/customize-webviews#cookie-sharing-and-sso-implications) With ASWebAuthenticationSession/SFSafariViewController, the cookies that were dropped by ests after signing in their app can be shared with the next web view that they open and they won't have to re-attach bearer AT in the Authentication header the second time. Is this something works for your scenario ?

**Comment by renep** [comment]
@swasti29 I think you don't get my problem. The authentication is not the problem. It works find. So the app is properly authenticated and REST-JSON requests work find, because here I can add the bearer token at the request header, and the request is passed throught the Azure AD App Proxy to the server. No problem here. 
The app I'm developing also loads HTML sites from server that are displayed WKWebView that the app show. The HTML page can be properly loaded, because for the request I can also add the bearer token.
But, and here is the problem: The HTML page contains additional resources like `<script src="script.js"... ` and when the WKWebView tries to load this resource, then it fails, because the WKWebView does append the bearer token to the request. So the WKWebView can not display this page properly.
It is also not possible to intercept these request the WKWebView makes. I have tried, but have not found a way.
The only solution at my point of view is, that the WKWebView has a proper cookie with the information needed to get through the Azure AD App Proxy. But I have not found any API at the MSAL libraray where I can request a Cookie that I can pass to the WKWebView.

**Comment by swasti29** [comment]
@renep 
MSAL does not support cookie based authentication.
The only suggestion i have is to resuse the cookie issued by the Azure(as given [here](https://learn.microsoft.com/en-us/entra/identity/authentication/concept-authentication-web-browser-cookies)) after signing in your app.

**Comment by juan-arias** [comment]
Closing for now, please feel free to reopen if there are any question related to this.

**Comment by renep** [comment]
I just wanted to add an comment here to the issue because I have found a solution for the problem. I have removed the Authenticaton token at the request was is loaded by the WKWebView. The outcome here is than that the login screen is dispayed in the WKWebView and the user has to login a second time. That is not nice but then the WKWebView gets the cookie. 
There is then an option to avoid the second login. When the [Enterise SSO](https://learn.microsoft.com/en-us/mem/intune/configuration/use-enterprise-sso-plug-in-ios-ipados-macos?pivots=macos) is configured, then the WKWebView can make the authentication without any user interaction, because the credentials are already present, because this is done after an successful MSAL authentication.
